File: platform/control-centre/src/main/scala/strata/server/lobby/controlcentre/repository/mapper/GameVariationRepository.scala
package strata.server.lobby.controlcentre.repository.mapper

import strata.server.lobby.controlcentre.model.GameVariation
import com.yazino.platform.model.PagedData

trait GameVariationRepository {
    def listFor(gameType: String): Map[BigDecimal, String]

    def findById(id: BigDecimal): Option[GameVariation]

    def findAll(page: Int = 0, pageSize: Int = 20): PagedData[GameVariation]

    def save(gameVariation: GameVariation): GameVariation

    def delete(gameVariationId: BigDecimal): Unit
}

File: platform/platform-grid/src/main/java/com/yazino/platform/repository/table/GameVariationRepository.java
package com.yazino.platform.repository.table;

import com.yazino.platform.model.table.Table;
import com.yazino.platform.table.GameVariation;

import java.math.BigDecimal;
import java.util.Set;

public interface GameVariationRepository {
    void refreshAll();

    GameVariation findById(BigDecimal id);

    BigDecimal getIdForName(String name,
                            String gameType);

    Set<GameVariation> variationsFor(String gameType);

    void loadTemplatesIfRequired();

    void populateProperties(Table table);
}
